src/schemes/boundary/open_boundary/system.jl

```diff
@@ -262,6 +259,31 @@ end
     return system
 end
 
+# Buffer particle is outside the boundary zone
+@inline function convert_particle!(system::OpenBoundarySPHSystem, fluid_system,
+                                   boundary_zone::BoundaryZone{BidirectionalFlow}, particle,
+                                   v, u, v_fluid, u_fluid)
+    particle_position = current_coords(u, system, particle) - boundary_zone.zone_origin
```

Comment:
```suggestion
    relative_position = current_coords(u, system, particle) - boundary_zone.zone_origin
```
Maybe a bit clearer?